### PR TITLE
add github workflow for publish

### PR DESCRIPTION
**Pull request by janstoff**

### CODE REVIEW

File: .github/workflows/publish_github_package.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      - name: Update package.json version
+        run: npm version ${{ steps.define_tag.outputs.PACKAGE_VERSION }} --no-git-tag-version
+
+      - name: Build
+        run: npm run build
+
+      - name: Publish
+        run: npm publish
+        env:
+          NODE_AUTH_TOKEN: ${{secrets.NPM_TOKEN}}
```
Comment (haoAddsearch) [inline .github/workflows/publish_github_package.yml:43]: How does `secrets.NPM_TOKEN` works? I didn't find the secret with this name in this repo.

File: .github/workflows/publish_github_package.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        id: define_tag
+        run: |
+          PACKAGE_VERSION=$(git describe --abbrev=0 --tags | sed 's/^v//')
+          echo "PACKAGE_VERSION=$PACKAGE_VERSION" >> $GITHUB_OUTPUT
+
+      - name: Update package.json version
+        run: npm version ${{ steps.define_tag.outputs.PACKAGE_VERSION }} --no-git-tag-version
+
+      - name: Build
+        run: npm run build
```
Comment (haoAddsearch) [inline .github/workflows/publish_github_package.yml:38]: Aren't we suppose to build already before creating the release?

File: .github/workflows/publish_github_package.yml (hunk 1 above)
Comment (janstoff) [reply .github/workflows/publish_github_package.yml:43]: As per slack.

File: .github/workflows/publish_github_package.yml (hunk 2 above)
Comment (janstoff) [reply .github/workflows/publish_github_package.yml:38]: No, we build before publish, but we need to set the version before build.

**Review by haoAddsearch — APPROVED** [review]

**Pull request merged**